### PR TITLE
Update Sar/Dar staticmethods to classmethods, other

### PR DESCRIPTION
**Pull request by LightArrowsEXE**

### CODE REVIEW

File: vstools/enums/other.py [hunk 1]
```diff
@@ -50,22 +50,32 @@ def string(self) -> str:
...
+        /,
+    ) -> DarSelf:
         width: int
         height: int
         sar: Sar | Literal[False]
 
         if isinstance(clip_width, vs.VideoNode):
+            from ..functions.check import check_variable_resolution
+
+            check_variable_resolution(clip_width, cls.from_size)  # type:ignore
```
[the remark this answers is not in the document]
Comment (LightArrowsEXE) [reply vstools/enums/other.py:77]: Not entirely sure what you mean by that, so will leave it to you.

File: vstools/enums/other.py [hunk 2]
```diff
@@ -72,9 +72,9 @@ def from_size(
         sar: Sar | Literal[False]
 
         if isinstance(clip_width, vs.VideoNode):
-            from ..functions.check import check_variable_resolution
+            from ..functions import check_variable_resolution
 
-            check_variable_resolution(clip_width, cls.from_size)  # type:ignore
+            assert check_variable_resolution(clip_width, cls.from_size)  # type:ignore
```
Comment (Setsugennoao) [inline vstools/enums/other.py:77]: assert isn't really needed for variable resolution, the typeguard doesn't do anything

**Review by Setsugennoao — CHANGES_REQUESTED** [review]

File: vstools/enums/other.py (hunk 2 above)
Comment (LightArrowsEXE) [reply vstools/enums/other.py:77]: remnant of when I had `check_variable` there before realising I only had to check the res

File: vstools/enums/other.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LightArrowsEXE) [reply vstools/enums/other.py]: Added in 68cd37291c9153c4efd868e1124eab61713ebecd.

**Pull request merged**